src/algorithms/standard/tensortranspose.cpp

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (C) 2006-2020  Music Technology Group - Universitat Pompeu Fabra
+ *
+ * This file is part of Essentia
+ *
+ * Essentia is free software: you can redistribute it and/or modify it under
+ * the terms of the GNU Affero General Public License as published by the Free
+ * Software Foundation (FSF), either version 3 of the License, or (at your
+ * option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the Affero GNU General Public License
+ * version 3 along with this program.  If not, see http://www.gnu.org/licenses/
+ */
+
+#include "tensortranspose.h"
+#include <sstream>
+
+using namespace essentia;
+using namespace standard;
+
+using namespace std;
+
+const char* TensorTranspose::name = "TensorTranspose";
+const char* TensorTranspose::category = "Standard";
+const char* TensorTranspose::description = DOC("This algorithm performs transpositions over the axes of a tensor.\n");
+
+  void TensorTranspose::configure() {
+    if (!parameter("permutation").isConfigured()) return;
+    _permutation = parameter("permutation").toVectorInt();
+
+    if (_permutation.size() != TENSORRANK) {
```

Comment:
These checks do not take into consideration the case of  having the same value repeated in the permutation vector.